RichString.h

```diff
@@ -15,21 +15,21 @@ in the source distribution for its full text.
 #define RichString_size(this) ((this)->chlen)
 #define RichString_sizeVal(this) ((this).chlen)
 
-#define RichString_begin(this) RichString (this); RichString_beginAllocated(this);
-#define RichString_beginAllocated(this) memset(&(this), 0, sizeof(RichString)); (this).chptr = (this).chstr;
-#define RichString_end(this) RichString_prune(&(this));
+#define RichString_begin(this) RichString (this); RichString_beginAllocated(this)
+#define RichString_beginAllocated(this) memset(&(this), 0, sizeof(RichString)); (this).chptr = (this).chstr
```

Comment:
`RichString_beginAllocated` now becomes *unsafe* cause one can append it, e.g. `RichString_beginAllocated(out) + 2;`. (I rather have an unnecessary semicolon than an unsafe macro.)
Maybe convert into an `static inline` function?